### PR TITLE
Fixing security issues in metering and monitoring dashboard api calls

### PR DESCRIPTION
**Pull request by Thanu**
This p/r is to fix to:
1. Add authentication logic in dashbaord jaggery files
2. Removing datasource param from request object in gadgets api calls.

### CODE REVIEW

File: extensions/das/artifacts/metering-dashboard/jaggery-files/applications.jag [hunk 1]
```diff
@@ -16,7 +16,57 @@
  */
 (function () {
     var log = new Log();
+
     var HTTP_INTERNAL_ERROR = 500;
```
Comment (imesh) [inline extensions/das/artifacts/metering-dashboard/jaggery-files/applications.jag]: HTTP_INTERNAL_ERROR has not been used. Can't we remove this variable?

Comment (Thanu) [inline extensions/das/artifacts/metering-dashboard/jaggery-files/applications.jag]: Fixed in commit fe38d24e4f3abef3400eb9c93f093000c2470371

File: extensions/das/artifacts/metering-dashboard/capps/ppaas-metering-service/GadgetMemberCount/Member_Count/js/gadgetconf.js [hunk 2]
```diff
@@ -16,7 +16,6 @@
 var gadgetConfig = {
     "id": "Member_Count",
     "title": "Member_Count",
-    "datasource": "MEMBER_COUNT",
```
Comment (imesh) [inline extensions/das/artifacts/metering-dashboard/capps/ppaas-metering-service/GadgetMemberCount/Member_Count/js/gadgetconf.js]: Why did we remove this variable? I understand that we needed to remove the table name from the request parameter. Is this something we added initially?

File: extensions/das/artifacts/metering-dashboard/capps/ppaas-metering-service/GadgetMemberCount/Member_Count/js/main.js [hunk 3]
```diff
@@ -13,7 +13,7 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-var datasource, type, columns, filter, maxUpdateValue;
```
Comment (imesh) [inline extensions/das/artifacts/metering-dashboard/capps/ppaas-metering-service/GadgetMemberCount/Member_Count/js/main.js]: What would be the reason for removing the filter variable?

Comment (Thanu) [inline extensions/das/artifacts/metering-dashboard/capps/ppaas-metering-service/GadgetMemberCount/Member_Count/js/main.js]: It is not used in the code.

File: extensions/das/artifacts/metering-dashboard/capps/ppaas-metering-service/GadgetMemberCount/Member_Count/js/gadgetconf.js (hunk 2 above)
Comment (Thanu) [inline extensions/das/artifacts/metering-dashboard/capps/ppaas-metering-service/GadgetMemberCount/Member_Count/js/gadgetconf.js]: Yes. It was there earlier and I used this variable to define the table name for the gadget

**Comment by ravihansa3000** [comment]
Why do we have authentication related logic in every Jaggery file? Better to include this from a common Jag file.

File: extensions/das/artifacts/metering-dashboard/jaggery-files/member-status.jag [hunk 4]
```diff
@@ -17,9 +17,55 @@
...
+        if (log.isDebugEnabled()) {
+            log.debug("Unable to instantiate AnalyticsWebServiceStub instance. This could be a CEP server. Returning");
+        }
+        return;
+    }
+    var serviceClient = analyticsStub._getServiceClient();
+    var options = serviceClient.getOptions();
+    options.setManageSession(true);
+
+    var authParam = request.getHeader(AUTHORIZATION_HEADER);
```
Comment (ravihansa3000) [inline extensions/das/artifacts/metering-dashboard/jaggery-files/member-status.jag]: How do we set this authorization header in client side?

Comment (imesh) [inline extensions/das/artifacts/metering-dashboard/jaggery-files/member-status.jag]: We do not need to explicitly set it. This is how standard DAS gadgets work. Once the end user log into the DAS portal, the client will receive a session cookie. This cookie includes a session token/JSESSIONID. On the next call from the client to the API will have this. As a result the API call will get authenticated.

Comment (ravihansa3000) [inline extensions/das/artifacts/metering-dashboard/jaggery-files/member-status.jag]: Thanks for the explanation. That cleared my doubts :)

**Comment by imesh** [comment]
Regarding moving authentication logic to one jaggery file and reusing, will try to merge this PR and do that modification. Thanks!

**Pull request merged**